src/Arch/OpenRISC/Aeon/AeonRewriter.cs

```diff
@@ -529,15 +567,19 @@ private void RewriteJr()
             var op0 = (Identifier)Op(0);
             if ((int)op0.Storage.Domain == 9)
             {
-                //$REVIEW: this is a weird encoding for a return
                 iclass = InstrClass.Transfer | InstrClass.Return;
                 m.Return(0, 0);
                 return;
             }
-            //$REVIEW: could be a call, r9 seems to be used as a link register.
             m.Goto(op0);
         }
 
+        private void RewriteJalr()
+        {
+            var target = (Identifier)Op(0);
```

Comment:
The cast is not needed here. `m.Call` accepts `Expression` which is a base class of `Identifier`.